component/pull-secret-sync-job.libsonnet

```diff
@@ -0,0 +1,166 @@
+local inv = kap.inventory();
+// The hiera parameters for the component
+local params = inv.parameters.openshift4_config;
+
+// Jobs need get,update,patch for secret pull-secret in namespace openshift-config
+// To ensure the unmanage job has the RBAC in place, all the RBAC objects are
+// also in sync-wave -10.
+local jobSA = kube.ServiceAccount('syn-cluster-pull-secret-manager') {
+  metadata+: {
+    annotations+: {
+      'argocd.argoproj.io/sync-wave': '-10',
+    },
+    namespace: 'openshift-config',
+  },
+};
+local jobRole = kube.Role('syn-cluster-pull-secret-manager') {
+  metadata+: {
+    annotations+: {
+      'argocd.argoproj.io/sync-wave': '-10',
+    },
+  },
+  rules: [
+    {
+      apiGroups: [ '' ],
+      resources: [ 'secrets' ],
+      verbs: [ 'get', 'update', 'patch' ],
+      resourceNames: [ 'pull-secret' ],
+    },
+  ],
+};
+local jobRoleBinding = kube.RoleBinding('syn-cluster-pull-secret-manager') {
+  metadata+: {
+    annotations+: {
+      'argocd.argoproj.io/sync-wave': '-10',
+    },
+  },
+  roleRef_: jobRole,
+  subjects_: [ jobSA ],
+};
+
+local cleanJob = kube.Job('syn-unmanage-cluster-pull-secret') {
+  metadata+: {
+    annotations+: {
+      // run before the default sync wave, but after creating the Job RBAC so
+      // that we unmanage the cluster pull secret before patching it.
+      'argocd.argoproj.io/sync-wave': '-9',
+      'argocd.argoproj.io/hook': 'Sync',
+      'argocd.argoproj.io/hook-delete-policy': 'HookSucceeded',
+    },
+  },
+  spec+: {
+    template+: {
+      spec+: {
+        serviceAccountName: jobSA.metadata.name,
+        containers_: {
+          clean: {
+            image: '%(registry)s/%(repository)s:%(tag)s' % params.images.kubectl,
+            command: [
+              'bash',
+              '-c',
+              'kubectl label secret pull-secret argocd.argoproj.io/instance-;' +
+              'kubectl annotate secret pull-secret kubectl.kubernetes.io/last-applied-configuration-;' +
+              'kubectl annotate secret pull-secret argocd.argoproj.io/sync-options-;',
+            ],
+          },
+        },
+      },
+    },
+  },
+};
+
+local syncScript = kube.Secret('syn-update-cluster-pull-secret-script') {
+  stringData: {
+    // The shell script reads the secret `pull-secret`, base64-decodes the
+    // value of `.dockerconfigjson`, processes it with jq and updates the
+    // secret with the result of the JQ script (see below).
+    'sync-secret.sh': |||
+      #!/bin/bash
+
+      pull_secret=$(
+          kubectl get secret pull-secret \
+          -o go-template='{{index .data ".dockerconfigjson"|base64decode}}'
+      )
+      patched_secret=$(
+        jq -cr '%(script)s' <<<"${pull_secret}"
+      )
+      kubectl -n openshift-config patch secret pull-secret \
+        -p "{\"data\": {\".dockerconfigjson\": \"$patched_secret\"}}"
+    ||| % {
+      // We generate a JQ script which processes the pull-secret contents from
```

Comment:
Took a while to understand. Tested with email `"github asd"@example.com`
```
{"auths":{"ghcr.io":{"auth":"[...]","email":"\"github asd\"@example.com"}}
```

Works 😊